core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1855,7 +1855,11 @@ object SnappySession extends Logging {
         evaluatePlan(df, session, key.sqlText, key)
       }
     }
```

Comment:
There could have been a small penalty of volatile read, I think, but since this will then do a ConcurrentMap lookup so then that will have volatile reads in any case which will flush cache lines.